File: development/NET/samples/server/DemoServer/CS/Creator.cs
using System;
using Softing.OPCToolbox.Server;

namespace DemoServer
{
	public class DemoCreator : Creator
	{
		#region Public Methods
		//---------------------

		public override AeAddressSpaceElement CreateInternalAeAddressSpaceElement(
			string aName,
			uint anUserData,
			uint anObjectHandle,
			uint aParentHandle)
		{
			return new DemoAeAddressSpaceElement(
				aName,
				anUserData,
				anObjectHandle,
				aParentHandle);
		}   //	end CreateAeAddressSpaceElement

		public override DaAddressSpaceRoot CreateDaAddressSpaceRoot()
		{
			return (DaAddressSpaceRoot)new DemoDaAddressSpaceRoot();
		}   //	end CreateDaAddressSpaceRoot

		public override DaTransaction CreateTransaction(
			EnumTransactionType transactionType,
			DaRequest[] requestList,
			uint sessionKey)
		{
			return (DaTransaction)new Transaction(
									transactionType,
									requestList,
									sessionKey);
		}   //	end CreateTransaction

		public virtual DemoDaAddressSpaceElement CreateBasicElement(String name)
		{
			return (DemoDaAddressSpaceElement)new BasicElement(name);
		}   //	end CreateBasicElement

		public virtual DemoDaAddressSpaceElement CreateBasicStaticElement(String name)
		{
			return (DemoDaAddressSpaceElement)new BasicStaticElement(name);
		}   //  end CreateBasicStaticElement

		public virtual DemoDaAddressSpaceElement CreateBasicStaticElement(System.Type type)
		{
			return (DemoDaAddressSpaceElement)new BasicStaticElement(type);
		}   //  end CreateBasicStaticElement

		public virtual DemoDaAddressSpaceElement CreateBasicDynamicElement(String name)
		{
			return (DemoDaAddressSpaceElement)new BasicDynamicElement(name);
		}   //  end CreateBasicDynamicElement

		public virtual DemoDaAddressSpaceElement CreateBasicDynamicElement(System.Type type)
		{
			return (DemoDaAddressSpaceElement)new BasicDynamicElement(type);
		}   //  end CreateBasicDynamicElement

		public virtual DemoDaAddressSpaceElement CreateSinusElement()
		{
			return (DemoDaAddressSpaceElement)new SinFunction();
		}   //  end CreateSinusElement

		public virtual DemoDaAddressSpaceElement CreateCosinusElement()
		{
			return (DemoDaAddressSpaceElement)new CosFunction();
		}   //  end CreateCosinusElement

		public virtual DemoDaAddressSpaceElement CreateTanElement()
		{
			return (DemoDaAddressSpaceElement)new TanFunction();
		}   //  end CreateTanElement

		public virtual DemoDaAddressSpaceElement CreateSimulationElement()
		{
			return (DemoDaAddressSpaceElement)new SimulationVariable();
		}   //  end CreateSimulationElement

		public virtual DemoDaAddressSpaceElement CreateSleepIntervalElement()
		{
			return (DemoDaAddressSpaceElement)new SleepIntervalVariable();
		}   //  end CreateSleepIntervalElement

		public virtual DemoDaAddressSpaceElement CreateTimeElement(String name)
		{
			return (DemoDaAddressSpaceElement)new TimeVariable(name);
		}   //  end CreateTimeElement

		public virtual DemoDaAddressSpaceElement CreateTimeElement(
			String name,
			TimeVariable.TimeType type,
			TimeVariable.TimeZone zone)
		{
			return (DemoDaAddressSpaceElement)new TimeVariable(name, type, zone);
		}   //  end CreateTimeElement

		public virtual DemoDaAddressSpaceElement CreateAlarmSimulationElement(
			String name,
			AlarmSimulation.AlarmType type)
		{
			return (DemoDaAddressSpaceElement)new AlarmSimulation(name, type);
		}   //  end CreateAlarmSimulationElement

		public virtual DemoDaAddressSpaceElement CreateKeyElement()
		{
			return (DemoDaAddressSpaceElement)new KeyVariable();
		}   //  end CreateKeyElement

		public virtual DemoAeAddressSpaceElement CreateDemoAeAddressSpaceElement()
		{
			return new DemoAeAddressSpaceElement();
		}   //  end CreateDemoAeAddressSpaceElement

		public override DaRequest CreateRequest(
			EnumTransactionType aTransactionType,
			uint aSessionHandle,
			DaAddressSpaceElement anElement,
			int aPropertyId,
			uint aRequestHandle)
		{
			return new Request(
				aTransactionType,
				aSessionHandle,
				anElement,
				aPropertyId,
				aRequestHandle);
		}   //	end CreateRequest
			//--
		#endregion

	}   //	end class MyCreator
}

File: development/C++/samples/server/DemoServer/Creator.cpp
#include "stdafx.h"
#include "Creator.h"


DaAddressSpaceRoot* DemoCreator::createDaAddressSpaceRoot(void)
{
	return (DaAddressSpaceRoot*) new DemoDaAddressSpaceRoot();
}   //  end createDaAddressSpaceRoot

DaTransaction* DemoCreator::createTransaction(
	EnumTransactionType aTransactionType,
	std::vector<DaRequest*>& aRequestList,
	unsigned long aSessionKey)
{
	return (DaTransaction*) new Transaction(aTransactionType, aRequestList, aSessionKey);
}   //  end createTransaction

DemoDaAddressSpaceElement* DemoCreator::createBasicElement(tstring name)
{
	return (DemoDaAddressSpaceElement*) new BasicElement(name);
}

DemoDaAddressSpaceElement* DemoCreator::createBasicStaticElement(tstring name)
{
	return (DemoDaAddressSpaceElement*) new BasicStaticElement(name);
}

DemoDaAddressSpaceElement* DemoCreator::createBasicStaticElement(VARENUM varType)
{
	return (DemoDaAddressSpaceElement*) new BasicStaticElement(varType);
}

DemoDaAddressSpaceElement* DemoCreator::createBasicDynamicElement(tstring name)
{
	return (DemoDaAddressSpaceElement*) new BasicDymanicElement(name);
}

DemoDaAddressSpaceElement* DemoCreator::createBasicDynamicElement(VARENUM varType)
{
	return (DemoDaAddressSpaceElement*) new BasicDymanicElement(varType);
}

DemoDaAddressSpaceElement* DemoCreator::createSinusElement()
{
	return (DemoDaAddressSpaceElement*) new SinFunction();
}

DemoDaAddressSpaceElement* DemoCreator::createCosinusElement()
{
	return (DemoDaAddressSpaceElement*) new CosFunction();
}

DemoDaAddressSpaceElement* DemoCreator::createTanElement()
{
	return (DemoDaAddressSpaceElement*) new TanFunction();
}

DemoDaAddressSpaceElement* DemoCreator::createSimulationElement()
{
	return (DemoDaAddressSpaceElement*) new SimulationVariable();
}

DemoDaAddressSpaceElement* DemoCreator::createSleepIntervalElement()
{
	return (DemoDaAddressSpaceElement*) new SleepIntervalVariable();
}

DemoDaAddressSpaceElement* DemoCreator::createTimeElement(tstring name)
{
	return (DemoDaAddressSpaceElement*) new TimeVariable(name);
}

DemoDaAddressSpaceElement* DemoCreator::createTimeElement(
	tstring varName,
	enum TimeVariable::TimeType type,
	enum TimeVariable::TimeZone timeZone)
{
	return (DemoDaAddressSpaceElement*) new TimeVariable(varName, type, timeZone);
}

DemoDaAddressSpaceElement* DemoCreator::createAlarmSimulationElement(
	tstring varName,
	enum AlarmSimulation::AlarmType type)
{
	return (DemoDaAddressSpaceElement*) new AlarmSimulation(varName, type);
}

DemoDaAddressSpaceElement* DemoCreator::createKeyElement()
{
	return (DemoDaAddressSpaceElement*) new KeyVariable();
}

DemoAeAddressSpaceElement* DemoCreator::createDemoAeAddressSpaceElement()
{
	return new DemoAeAddressSpaceElement();
}

DaRequest* DemoCreator::createRequest(
	EnumTransactionType aTransactionType,
	unsigned long aSessionHandle,
	DaAddressSpaceElement* pAnElement,
	int aPropertyId,
	unsigned long aRequestHandle)
{
	return new Request(aTransactionType, aSessionHandle, pAnElement, aPropertyId, aRequestHandle);
}   //  end createRequest
